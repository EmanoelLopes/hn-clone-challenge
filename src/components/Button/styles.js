import styled from 'styled-components';

export const ButtonDefault = styled.button`
  background-color: ${props => !props.default ? '#fff' : '#e74c3c'};
  border: 1px solid #ddd;
  color: #333;
  cursor: pointer;
  font-size: 20px;
  padding: .5rem .75rem;

  &:hover {
    background-color: #eee;
  }

  &.primary {
    background-color: #e74c3c;
    border: 1px solid #e74c3c;
    border-radius: 4px;
    color: #fff;
    font-weight: 500;

    &:hover {
      opacity: .85;
    }

    span {
      font-size: 16px;
      margin: 0 .5rem;
      font-weight: 700;
    }
  }
`;