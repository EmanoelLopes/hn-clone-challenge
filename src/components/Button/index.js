import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { ButtonDefault } from './styles';

function Button ({ icon, label, type, click }) {
  return (
    <ButtonDefault
      className={(type) ? `button ${type}` : 'button default'}
      arial-label={label}
      onClick={click}>
      { (icon) ? <FontAwesomeIcon icon={icon} /> : null }
      { (label) ? <span>{ label }</span> : null }
    </ButtonDefault>
  );
};

Button.propTypes = {
  type: PropTypes.string,
  icon: PropTypes.object,
  label: PropTypes.string,
  click: PropTypes.func
};

export default Button;