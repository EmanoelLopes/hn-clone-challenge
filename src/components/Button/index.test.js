import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from '../Button';

configure({ adapter: new Adapter() });

const props = {
  icon: {}, 
  label: 'Ok',
  type: 'primary'
};

describe('<Button />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });

  it('test click event', () => {
    const mockCallBack = jest.fn();

    const wrapper = shallow((<Button {...props} click={mockCallBack}>Ok!</Button>));
    wrapper.find('.button').first().simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});

