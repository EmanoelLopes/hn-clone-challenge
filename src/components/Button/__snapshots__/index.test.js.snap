// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "arial-label": undefined,
      "children": Array [
        null,
        null,
      ],
      "className": "button default",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "arial-label": undefined,
        "children": Array [
          null,
          null,
        ],
        "className": "button default",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
